server/routes/user.js

```diff
@@ -2,13 +2,25 @@ import express from 'express';
 import UserController from '../controllers/UserController';
 import middlewares from '../middlewares';
 
-const { favouriteAuthor, unfavouriteAuthor } = UserController;
-const { verifyToken, authorizeUser, AuthorValidator: { favAuthorValidation } } = middlewares;
+const {
+  favouriteAuthor, unfavouriteAuthor, favoriteBook, unfavoriteBook
+} = UserController;
+const {
+  verifyToken, authorizeUser, AuthorValidator: { favAuthorValidation },
+  BookValidator: { FavoriteBookValidation }
+} = middlewares;
 
 const user = express.Router();
 const FAVOURITES_BASE_URL = '/favourites';
 
+// Favorite Author
 user.post(`${FAVOURITES_BASE_URL}`, verifyToken, authorizeUser(['patron']), favAuthorValidation(), favouriteAuthor);
+// Unfavorite Author
 user.delete(`${FAVOURITES_BASE_URL}`, verifyToken, authorizeUser(['patron']), favAuthorValidation(), unfavouriteAuthor);
 
+// Favorite Book
+user.post('/favorites/book/:bookId', verifyToken, authorizeUser(['patron']), FavoriteBookValidation(), favoriteBook);
+// Unfavorite Book
+user.delete('/favorites/book/:bookId', verifyToken, authorizeUser(['patron']), FavoriteBookValidation(), unfavoriteBook);
```

Comment:
Delete should be a query, not a param

Reply:
Query how?

Reply:
You are correct, it can be used both has a param or query string